spec/e2e/encrypt_spec.rb

```diff
@@ -0,0 +1,159 @@
+      end
+
+      it 'should encrypt an integer' do
+        payload = 1
+        encrypted = client.encrypt(payload)
+        decrypted = client.decrypt(encrypted)
+        expect(decrypted).to eq(payload)
+      end
+
+      it 'should decrypt an integer with a permitting role' do
+        payload = 1
+        encrypted = client.encrypt(payload, "permit-all")
+        decrypted = client.decrypt(encrypted)
+        expect(decrypted).to eq(payload)
+      end
+
+      it 'should error decrypting an integer with a denying role' do
+        payload = 1
+        encrypted = client.encrypt(payload, 'deny-all')
+        expect { client.decrypt(encrypted) }.to raise_error(Evervault::Errors::EvervaultError)
+      end
+
+      it 'should encrypt a float' do
+        payload = 1.0
+        encrypted = client.encrypt(payload)
+        decrypted = client.decrypt(encrypted)
+        expect(decrypted).to eq(payload)
+      end
+
+      it 'should decrypt a float with a permitting role' do
+        payload = 1.0
+        encrypted = client.encrypt(payload, 'permit-all')
+        decrypted = client.decrypt(encrypted)
+        expect(decrypted).to eq(payload)
+      end
+
+      it 'should error decrypting a float with a denying role' do
+        payload = 1.0
+        encrypted = client.encrypt(payload, 'deny-all')
+        expect { client.decrypt(encrypted) }.to raise_error(Evervault::Errors::EvervaultError)
+      end
+
+      it 'should encrypt a true bool' do
+        payload = true
+        encrypted = client.encrypt(payload)
+        decrypted = client.decrypt(encrypted)
+        expect(decrypted).to eq(payload)
+      end
+
+      it 'should decrypt a true bool with a permitting role' do
+        payload = true
+        encrypted = client.encrypt(payload, 'permit-all')
+        decrypted = client.decrypt(encrypted)
+        expect(decrypted).to eq(payload)
+      end
+
+      it 'should error decrypting a true bool with a denying role' do
+        payload = true
+        encrypted = client.encrypt(payload, 'deny-all')
+        expect { client.decrypt(encrypted) }.to raise_error(Evervault::Errors::EvervaultError)
+      end
+
+      it 'should encrypt a false bool' do
+        payload = false
+        encrypted = client.encrypt(payload)
+        decrypted = client.decrypt(encrypted)
+        expect(decrypted).to eq(payload)
+      end
+
+      it 'should decrypt a false bool with a permitting role' do
+        payload = false
+        encrypted = client.encrypt(payload, 'permit-all')
+        decrypted = client.decrypt(encrypted)
+        expect(decrypted).to eq(payload)
+      end
+
+      it 'should error decrypting a false bool with a denying role' do
+        payload = false
+        encrypted = client.encrypt(payload, 'deny-all')
+        expect { client.decrypt(encrypted) }.to raise_error(Evervault::Errors::EvervaultError)
+      end
+
+      it 'should encrypt a hash' do
+        payload = { 'hello' => 'world' }
```

Comment:
Should we add a few more elements to the hash to ensure we properly iterate through it while encrypting?

```
{
  "string": "apple",
  "integer": 12345,
  "float": 123.45,
  "true": true,
  "false": false,
  "array": ["apple", 12345, 123.45, true, false]
}
```

Reply:
Good call! Will update now